README.md

```diff
@@ -126,6 +126,25 @@ Update the git repo and checkout the tag
 sgit update redis
 ```
 
+
+## Fetch a repo
+
+If you want to `git fetch` all or a subset of git repos in your config then you can use the `sgit fetch` command. The benefit of doing a fetch is that you can fetch home all changes to a set of git repos but you do not have to update and move each repo to a new commit. In general git operations, it is always more safe to run `git fetch` before you do a checkout or `git pull` to update your local cloned repos. This allows you to inspect the changes incomming before commiting to pulling them.
+
+Sgit fetch command supports the selection of either all repos or a subset of repos. The fetch command will never prompt the user asking if they want to do a update as fetch is considered a non-descrutive command.
+
+```bash
+# Fetch all repos implicitly
+sgit fetch
+
+# Fetch all repos explicltly
+sgit fetch all
```

Comment:
Scenario: you have a repo named "all"

Proposed solution:

Use the common practice of using a space-double-dash-space "` -- `" to tell the command line parser to end parsing commands and use the following arguments literally. So, if invoked like so `sgit fetch -- all` then it would try to update a repository named "all" instead of updating _all_ repositories.

Of course any named repository could follow the space-double-dash-space and it would do the right thing, e.g. `sgit fetch -- pykwalify` would be analogous to `sgit fetch pykwalify`.

Reply:
@holmboe I think this is already supported and works. If you try this setup

```
repos:
  all:
    clone-url: git@github.com:Grokzen/pykwalify.git
    revision:
      branch: master
  pykwalify:
    clone-url: git@github.com:Grokzen/pykwalify.git
    revision:
      branch: master
```

and run `sgit update` first then `sgit fetch all` 

```
(sgit) ➜  sgit git:(f/fetch-command) ✗ sgit fetch all
DEBUG: Repo fetch - ['all']
INFO: repos: ['all', 'pykwalify']
Fetching git repo 'all'
Fetching completed for repo 'all'
 - Fetch result: origin/master
 - Fetch result: origin/feature/rebuild-include
Fetching git repo 'pykwalify'
Fetching completed for repo 'pykwalify'
 - Fetch result: origin/master
 - Fetch result: origin/feature/rebuild-include
Fetching for all repos completed
```

the logic is that `all` is a reserved keyword that will only ever mean `all`.

Also technically `all` maybe should not be an option to input really as it is only a internal mechanism to deal with cases where you do not input anything into the optional block `[<repos> ...]` and maybe that logic should be reworked a bit because if you do `sgit fetch` that implies all repos.

In the end i am probably more in favor of removing the `all` as a reserved keyword and always interprete `all` as a repo name and avoid implementing the `--` hack around this.

Reply:
Using ` -- ` is perhaps not a hack in that sense. It is the de facto way to distinguish between a command (e.g. `fetch all`) and an argument (`fetch -- all`).

Another alternative is to implement `--all` as a command line switch instead of using a word that could have multiple meanings. But implementing `all` as a reserved keyword for all repositories is arguably not a bad UX at all.

Given the two alternatives, I am in favor of implementing ` -- ` to distinguish the two is a good way forward.

Reply:
I reverted out from the keyword support of `all` and now all repo names will be interpreated as repo names and the only way to indicate you want to update or fetch all repos is to simply write `sgit update` and `sgit fetch` and there is not really a huge need to have both `sgit update` and `sgit update all` as both of them would do the exact same thing. I think this way will simplify all commands we have implemented now and it is more in line with what git does